osu.Game/Extensions/ModelExtensions.cs

```diff
@@ -137,20 +137,14 @@ private static bool matchesOnlineID(this IHasOnlineID<int>? instance, IHasOnline
             return instance.OnlineID.Equals(other.OnlineID);
         }
 
-        private static readonly char[] invalid_filename_characters = Path.GetInvalidFileNameChars()
-                                                                         // Backslash is added to avoid issues when exporting to zip.
-                                                                         // See SharpCompress filename normalisation https://github.com/adamhathcock/sharpcompress/blob/a1e7c0068db814c9aa78d86a94ccd1c761af74bd/src/SharpCompress/Writers/Zip/ZipWriter.cs#L143.
-                                                                         .Append('\\')
-                                                                         .ToArray();
+        private static Regex invalid_filename_chars = new(@"(?!$)[^A-Za-z0-9_()[\] -]", RegexOptions.Compiled);
```

Comment:
```suggestion
        private static readonly Regex invalid_filename_chars = new Regex(@"(?!$)[^A-Za-z0-9_()[\] -]", RegexOptions.Compiled);
```

We don't use target-typed `new` in this repository.

Reply:
Fixed.